src/mutation_probabilities.rs

```diff
@@ -0,0 +1,153 @@
+use crate::mutation::MutationType;
+use rand::prelude::{Rng, ThreadRng};
+use std::convert::TryFrom;
+
+/// Describes the relative probabilites of different structural mutations.
+///
+/// Use `MutationProbabilitiesFloat` for a more ergonomic API that allows for entry as f64s.
+///
+/// Note: fits in a single 64bit register. Copy, don't reference.
+#[derive(Copy, Clone)]
+pub struct MutationProbabilities(u64);
```

Comment:
Yes, agree, or we may honestly throw this out, it's sort of a dumb addition I made.

Reply:
I think it's a good API addition, so it should stay. The internals of it can be replaced with this function from `rand` though: https://docs.rs/rand/latest/rand/seq/trait.SliceRandom.html#tymethod.choose_weighted

Reply:
Replaced, I think it is now guaranteed to be sound.  No tests yet.